middle_end/flambda/lift_let_to_initialize_symbol.ml

```diff
@@ -65,24 +66,33 @@ let rec accumulate ~substitution ~copied_lets ~extracted_lets
     }
   | Let { var; defining_expr = Expr (Var alias); body; _ }
   | Let_rec ([var, Expr (Var alias)], body) ->
+    let layouts =
+      Layouts.add layouts var (Layouts.find layouts alias)
+    in
     let alias =
       match Variable.Map.find alias substitution with
       | exception Not_found -> alias
       | original_alias -> original_alias
     in
     accumulate
+      ~layouts
       ~substitution:(Variable.Map.add var alias substitution)
       ~copied_lets
       ~extracted_lets
       body
   | Let { var; defining_expr = named; body; _ }
   | Let_rec ([var, named], body)
     when should_copy named ->
+      let layout = Flambda.result_layout_named ~layouts named in
```

Comment:
Here (and in the other `Let_rec` cases), shouldn't we use `Lambda.layout_letrec` instead? It is probably safe to do this when there is a single let-rec variable thanks to the rec-check, but it seems easier and less brittle to just use `layout_letrec` instead.

Reply:
Yes it would be nicer, but this should still be correct, and would require to duplicate some code to split the let and letrec cases